Tests/DatadogTests/Helpers/EquatableInTests.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Utility protocol adding `Equatable` conformance to any arbitrary type.
+/// The equatabiliity is determined based on comparing type mirrors and values.
+protocol EquatableInTests: Equatable {}
+
+extension EquatableInTests {
```

Comment:
💡 This new implementation of `EquatabeInTests` is way smarter than the previous one. The old one was just comparing `String(describing: lhs)` vs `String(describing: rhs)`, which was not working with unordered collections like `[String: Encodable]` or `Set<AnyHashable>`. Now, the comparison looks into each value's `Mirror` and decides on the equality using reflection 👌.